provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/issue/BlackDuckProviderIssueHandler.java

```diff
@@ -36,29 +38,28 @@ public BlackDuckProviderIssueHandler(Gson gson, BlackDuckApiClient blackDuckApiC
         this.issueService = issueService;
     }
 
-    public void createOrUpdateBlackDuckIssue(BlackDuckProviderIssueModel issueModel, String bomComponentVersionIssuesUrl, String projectVersionUrl) throws IntegrationException {
+    public void createOrUpdateBlackDuckIssue(BlackDuckProviderIssueModel issueModel, @Nullable String bomComponentVersionIssuesUrl, String projectVersionUrl) throws IntegrationException {
         Optional<ProjectVersionIssuesView> optionalExistingIssue = retrieveExistingIssue(projectVersionUrl, issueModel.getKey());
 
         Date currentDate = Date.from(Instant.now());
         IssueRequest issueRequestModel = createIssueRequestModel(issueModel);
 
-        HttpMethod httpMethod = HttpMethod.POST;
-        HttpUrl requestUri = new HttpUrl(bomComponentVersionIssuesUrl);
-
         if (optionalExistingIssue.isPresent()) {
             ProjectVersionIssuesView existingIssue = optionalExistingIssue.get();
             issueRequestModel.setIssueDescription(existingIssue.getIssueDescription());
             issueRequestModel.setIssueCreatedAt(existingIssue.getIssueCreatedAt());
             issueRequestModel.setIssueUpdatedAt(currentDate);
 
             // The request uri should point at the specific issue for PUT requests
-            httpMethod = HttpMethod.PUT;
-            requestUri = existingIssue.getHref();
-        } else {
+            HttpUrl requestUri = existingIssue.getHref();
+            performRequest(requestUri, HttpMethod.PUT, issueRequestModel);
+        } else if (null != bomComponentVersionIssuesUrl) {
             issueRequestModel.setIssueCreatedAt(currentDate);
             issueRequestModel.setIssueUpdatedAt(null);
+
+            HttpUrl requestUri = new HttpUrl(bomComponentVersionIssuesUrl);
+            performRequest(requestUri, HttpMethod.POST, issueRequestModel);
```

Comment:
These changes are to address an issue where we were trying to update Black Duck issues for components that have been deleted. We don't need to update issues for components that have been deleted, Black Duck removes them automatically.